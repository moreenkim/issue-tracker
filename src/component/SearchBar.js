import React from 'react';
import { Input } from 'antd';
const { Search } = Input;

const SearchBar = (props) => {
  const { enterButtonText, placeholderText, onSearch, style } = props;
  return (
    <Search
      placeholder={placeholderText}
      allowClear
      enterButton={enterButtonText}
      size="large"
      onSearch={onSearch}
      style={style || {}}
    />
  );
};

export default SearchBar;
