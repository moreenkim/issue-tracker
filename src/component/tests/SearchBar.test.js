// Test SearchBar
import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import SearchBar from '../SearchBar';

describe('components/SearchBar', () => {
  const props = {
    enterButtonText: 'Search',
    placeholderText: 'Enter repository url e.g. owner/repository-name',
    onSearch: () => {},
  };
  it('renders without crashing', () => {
    shallow(<SearchBar {...props} />);
  });

  it('renders SearchBar correctly', () => {
    const wrapper = mount(<SearchBar {...props} />);
    expect(toJson()).toMatchSnapshot();
    wrapper.unmount();
  });
});
