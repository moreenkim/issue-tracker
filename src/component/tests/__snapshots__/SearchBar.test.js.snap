// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/SearchBar renders SearchBar correctly 1`] = `undefined`;
